mat64/io_386.go

```diff
@@ -0,0 +1,11 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat64
+
+import "math"
+
+const (
+	maxInt = math.MaxInt32
```

Comment:
Rather, why not use `const maxInt = int(^uint(0) >> 1)` directly in `UnmarshallBinary` instead of package-level const?